### PR TITLE
Feat(gateway discovery): Add config for gateway readiness check intervals and timeouts

### LINKED ISSUE #6425: Add configuration of Gateway Discovery related timeouts
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this enhancement require public documentation?

- [X] I have added an Acceptance Criteria item for adding and/or adjusting public documentation (if applicable)

### Problem Statement

Timeouts used in Gateway Discovery implementation (e.g. https://github.com/Kong/kubernetes-ingress-controller/blob/8593248e3214144fa6a6a83d21d4448cdac3d70c/internal/clients/manager.go#L19-L20 and https://github.com/Kong/kubernetes-ingress-controller/blob/b85dbb1fc36e7f24423447a91c1233144d7d6271/internal/clients/readiness.go#L16-L17) are set to unconfigurable values.

### Proposed Solution

- Implement CLI flags to allow configuration of
  - readiness check timeout
  - client readiness reconciliation period
- Implement a cross check functionality which will verify that check timeout is not bigger than reconciliation period

### Additional information

_No response_

### Acceptance Criteria

- [x] As a user I can configure readiness check timeout and client readiness reconciliation period
- [x] Documentation for those 2 CLI flags/envs is added

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

Add flags `--gateway-discovery-readiness-check-interval` and `--gateway-discovery-readiness-check-timeout` to configure reconciliation interval and timeout of readiness checks.

**Which issue this PR fixes**:

fixes #6425 
**Special notes for your reviewer**:

Naming problem: should we change `ReconciliationInterval` to `ReconciliationPeriod` to align with other `***Period`s?

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/manager/config.go [hunk 1]
```diff
@@ -63,21 +64,25 @@ type Config struct {
...
+	APIServerBurst                         int
+	APIServerCAData                        []byte
+	APIServerCertData                      []byte
+	APIServerKeyData                       []byte
+	MetricsAddr                            string
+	ProbeAddr                              string
+	KongAdminURLs                          []string
+	KongAdminSvc                           OptionalNamespacedName
+	GatewayDiscoveryDNSStrategy            cfgtypes.DNSStrategy
+	GatewayDiscoveryReadinessCheckInterval time.Duration // REVIEW(naming): the name seems too long. Any idea to shorten it?
```
Comment (pmalek) [inline internal/manager/config.go:78]: I don't mind this being long as it's descriptive and it's clear what it's used for.

File: internal/manager/config_validation.go [hunk 2]
```diff
@@ -122,6 +126,22 @@ func (c *Config) validateFallbackConfiguration() error {
 	return nil
 }
 
+func (c *Config) validateGatewayDiscovery() error {
+	// Skip validation if gateway discovery is not enabled.
+	if _, ok := c.KongAdminSvc.Get(); !ok {
+		return nil
+	}
+	// REVIEW: Should the reconciliation interval has a miniumum value and should it be the same as default interval?
```
Comment (pmalek) [inline internal/manager/config_validation.go:134]: I believe the interval should have a minimum. I would suggest something like 3s. WDYT?

Comment (randmonkey) [reply internal/manager/config_validation.go:134]: 3s that is same as default config sync interval seems good.

File: internal/clients/readiness.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ import (
 )
 
 const (
-	readinessCheckTimeout = 5 * time.Second
+	DefaultReadinessCheckTimeout = 5 * time.Second
```
Comment (pmalek) [inline internal/clients/readiness.go:17]: This should have a comment IMHO. Linter should enforce it I believe, no sure why it doesn't.

**Review by pmalek — APPROVED** [review]
1 nit about adding a comment to a `const`. Other than that this looks good 👍

File: internal/clients/readiness.go (hunk 3 above)
Comment (randmonkey) [reply internal/clients/readiness.go:19]: Added comments.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**